tests/src/Functional/ErrandServiceListTest.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\helfi_tpr\Functional;
+
+use Drupal\Tests\helfi_api_base\Functional\MigrationTestBase;
+use Drupal\Tests\helfi_api_base\Traits\ApiTestTrait;
+
+/**
+ * Tests entity list functionality.
+ *
+ * @group helfi_tpr
+ */
+class ErrandServiceListTest extends MigrationTestBase {
+
+  use ApiTestTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'views',
+    'helfi_tpr',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'stark';
+
+  /**
+   * Tests collection route (views).
+   */
+  public function testList() : void {
```

Comment:
This whole thing could probably be refactored into a generic `ListTestBase` class that has `testList()` method. Then make all `*ListTest`s to extend that class with some way to pass admin url and required permissions to that method.